week_2/dagster_ucr/project/week_2.py

```diff
@@ -5,26 +5,47 @@
 from dagster_ucr.resources import mock_s3_resource, redis_resource, s3_resource
 
 
-@op
-def get_s3_data():
-    pass
+@op(
+    out={"stocks": Out(dagster_type=List[Stock])},
+    required_resource_keys={"s3"},
+    tags={"kind": "s3"},
+    description="Get a list of stocks from an S3 file",
+)
+def get_s3_data(context):
+    stocks = list()
+    key = context.op_config["s3_key"]
```

Comment:
The `config_schema` isn't yet configured in this op. You would need this to access the `s3_key` config via context from inside the op.

Reply:
The config schema is created below (lines 53 & 73) and passed to the `op` in the `pipeline` calls that Dennis shared with us (78 & 84). The config parameter is optional in the decorator since the `context` can access the schema.  I think it might have been required in week 1 because we were using `config.yaml`. But in this week's exercise you can see in Dagit that the `op` knows what the `s3_key` is even if it's not set as a parameter. 
<img width="734" alt="image" src="https://user-images.githubusercontent.com/47192430/187092092-d9eb2c28-b917-4ba2-b00f-abba86d4c040.png">

Reply:
wow, and it works 😄 !  Indeed that possibility is mentioned in the [docs](https://docs.dagster.io/concepts/configuration/config-schema). Quite confusing for me, though 😄

Reply:
Dennis has kept it in all the examples, so I'm thinking it's probably best practice to have it though. Maybe to help document what variables are being used in the function, perhaps?